config/initializers/unify_google_compute.rb

```diff
@@ -0,0 +1,202 @@
+              f.write(params[:atmo_user_key].public_key)
+              f
+            end
+            server = start_server(params.merge(key_path: key_file.path))
+          else
+            server = start_server(params)
+          end
+
+          server
+        end
+
+        def destroy(id_at_site)
+          service.delete_server(id_at_site, ZONE)
+        end
+
+        private
+
+        def create_disk(params)
+          disk_params = {
+            name: params[:name],
+            # think about taking it from flavor or user requirement.
+            # Flavor disk is currently set to 0 so this also need to be
+            # rethinked.
+            size_gb: 10,
+            zone_name: params[:zone],
+            source_image: params[:image_id],
+            autoDelete: true
+          }
+          disk = service.disks.create(disk_params)
+          disk.wait_for { disk.ready? }
+
+          disk
+        end
+
+        def start_server(params)
+          server_params = {
+            name: params[:name],
+            machine_type: params[:flavor_id],
+            zone_name: params[:zone],
+            username: params[:username],
+            public_key_path: params[:key_path],
+            disks: [params[:disk]]
+          }
+
+          create_orig(server_params)
+        end
+      end
+
+      class Server
+        def image_id
+          disk_id = bootable_disk['source'].split('/').last
+          service.disks.get(disk_id).image_id
+        end
+
+        def task_state
+        end
+
+        def flavor_id
+          machine_type.split('/').last
+        end
+
+        def created
+          Time.parse(creation_timestamp)
+        end
+
+        def id
+          name
+        end
+
+        def stop
+          raise Atmosphere::UnsupportedException,
+                'Fog des not support google compute stop action'
+        end
+
+        def pause
+          raise Atmosphere::UnsupportedException,
+                'Google compute des not support pause action'
+        end
+
+        def suspend
+          raise Atmosphere::UnsupportedException,
+                'Google compute des not support suspend action'
+        end
+
+        def start
+          raise Atmosphere::UnsupportedException,
+                'Fog des not support google compute start action'
+        end
+
+        private
+
+        def bootable_disk
+          disks.detect { |d| d['boot'] }
+        end
+      end
+
+      class Disk
+        def image_id
+          source_image.split('/').last
+        end
+      end
+
+      class Real
+        def create_tags_for_vm(server_id, tags_map)
+          server = servers.get(server_id)
+          existing_tags = server.tags['items'] || []
+
+          clean = lambda { |s| s.downcase.gsub(/[^a-z0-9]/, '-') }
+          tags = tags_map.map { |k, v| "#{clean.call(k)}-#{clean.call(v)}" }
+
+          server.set_tags(existing_tags + tags)
+        end
+
+        def import_key_pair(is_at_site, public_key)
```

Comment:
Unused method argument - `is_at_site`. If it's necessary, use `_` or `_is_at_site` as an argument name to indicate that it won't be used. You can also write as `import_key_pair(*)` if you want the method to accept any arguments but don't care about them.<br>Unused method argument - `public_key`. If it's necessary, use `_` or `_public_key` as an argument name to indicate that it won't be used. You can also write as `import_key_pair(*)` if you want the method to accept any arguments but don't care about them.